db/migrate/20220506221447_set_pm_tags_allowed_for_groups_default.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class SetPmTagsAllowedForGroupsDefault < ActiveRecord::Migration[7.0]
+  def up
+
+    # if the old SiteSetting of `allow_staff_to_tag_pms` was set to true, update the new SiteSetting of
+    # `pm_tags_allowed_for_groups` default to include the staff group
+    allow_staff_to_tag_pms = DB.query_single("SELECT value FROM site_settings WHERE name = 'allow_staff_to_tag_pms'").first
+
+    # Dynamically sets the default value
+    if allow_staff_to_tag_pms == "t"
+      default = []
+      Group::AUTO_GROUPS.select do |group_key, id|
```

Comment:
> Migrations that lean on application code tend to be extremely brittle. A few years out a site setting may be removed a user field changed, or even worse, the method name could remain but the semantics change leading to subtle migration breakages.

Quoting our internal development lore here but we need to avoid relying on application code in migrations.